bin/check-beanstalkd.rb

```diff
@@ -103,26 +104,44 @@ def run
     ok
   end
 
+  JOB_STATES = [:ready, :urgent, :buried].freeze
+
   def check_queues(stats)
     msg = []
     crits = {}
     warns = {}
 
-    [:ready, :urgent, :buried].each do |task|
-      tasks = stats.send("current_jobs_#{task}".to_sym)
+    JOB_STATES.each do |job_state|
+      jobs = stats.send("current_jobs_#{job_state}".to_sym)
 
-      if tasks > config[task][1]
-        crits[task] = tasks
-        msg << task.to_s + " queue has #{tasks} items"
+      if jobs > config[job_state][1]
+        crits[job_state] = jobs
+        msg << job_state.to_s + " queue has #{jobs} items"
         next
       end
 
-      if tasks > config[task][0]
-        warns[task] = tasks
-        msg << task.to_s + " queue has #{tasks} items"
+      if jobs > config[job_state][0]
+        warns[job_state] = jobs
+        msg << job_state.to_s + " queue has #{jobs} items"
       end
     end
 
     [warns, crits, msg]
   end
+
+  def tube_stats
+    acquire_beanstalkd_connection.tubes[config[:tube].to_s].stats
+  rescue Beaneater::NotFoundError
+    empty_queue_stats
```

Comment:
I think there should be an option controlling what happens here. For example you might want to crit, warn, or ignore the failure. It all depends on the use case.

Reply:
something like:
```ruby
rescue Beaneater::NotFoundError => e
  if config[:alert_on_missing_tube] == 'critical'
    critical "Bean was not found: #{e}"
  elsif config[:alert_on_missing_tube] == 'warning'
    warning "Bean was not found: #{e}"
  else
    empty_queue_stats
  end
```

Reply:
Good point, will adjust accordingly.